[core]/es_extended/server/main.lua

```diff
@@ -122,252 +122,211 @@ if not Config.Multichar then
-					end
+	local xPlayer = CreateExtendedPlayer(playerId, identifier, userData.group, userData.accounts, userData.inventory, userData.weight, userData.job, userData.loadout, userData.playerName, userData.coords, userData.metadata)
+    ESX.Players[playerId] = xPlayer
+    Core.playersByIdentifier[identifier] = xPlayer
 
-					table.insert(userData.loadout,
-						{
-							name = name,
-							ammo = weapon.ammo,
-							label = label,
-							components = weapon.components,
-							tintIndex = weapon.tintIndex
-						})
-				end
-			end
-		end
-	end
+    -- Identity
+	if result.firstname and result.firstname ~= '' then 
+		userData.firstName = result.firstname
+		userData.lastName = result.lastname
 
-	-- Position
-	userData.coords = json.decode(result.position) or Config.DefaultSpawns[math.random(#Config.DefaultSpawns)]
+        xPlayer.set('firstName', result.firstname)
+		xPlayer.set('lastName', result.lastname)
 
-	-- Skin
-	if result.skin and result.skin ~= '' then
-		userData.skin = json.decode(result.skin)
-	else
-		if userData.sex == 'f' then
-			userData.skin = { sex = 1 }
-		else
-			userData.skin = { sex = 0 }
-		end
-	end
+		userData.playerName = userData.firstName .. ' ' .. userData.lastName
+        xPlayer.name = userData.playerName
 
-	-- Identity
-	if result.firstname and result.firstname ~= '' then
-		userData.firstname = result.firstname
-		userData.lastname = result.lastname
-		userData.playerName = userData.firstname .. ' ' .. userData.lastname
 		if result.dateofbirth then
 			userData.dateofbirth = result.dateofbirth
+            xPlayer.set('dateofbirth', result.dateofbirth)
 		end
 		if result.sex then
 			userData.sex = result.sex
+            xPlayer.set('sex', result.sex)
 		end
 		if result.height then
 			userData.height = result.height
+            xPlayer.set('height', result.height)
 		end
-	end
+    end 
 
-	if result.metadata and result.metadata ~= '' then
-		local metadata = json.decode(result.metadata)
-		userData.metadata = metadata
-	end
-
-	local xPlayer = CreateExtendedPlayer(playerId, identifier, userData.group, userData.accounts, userData.inventory, userData.weight, userData.job, userData.loadout, userData.playerName, userData.coords, userData.metadata)
-	ESX.Players[playerId] = xPlayer
-	Core.playersByIdentifier[identifier] = xPlayer
-
-	if userData.firstname then
-		xPlayer.set('firstName', userData.firstname)
-		xPlayer.set('lastName', userData.lastname)
-		if userData.dateofbirth then
-			xPlayer.set('dateofbirth', userData.dateofbirth)
-		end
-		if userData.sex then
-			xPlayer.set('sex', userData.sex)
-		end
-		if userData.height then
-			xPlayer.set('height', userData.height)
-		end
-	end
-	--saved player health and armor in metadata
-	local ped = GetPlayerPed(xPlayer.source)
-	if ped then
+    -- Save player health and armor in metadata
+    local ped = GetPlayerPed(xPlayer.source) 
+    if ped then
 		xPlayer.setMeta('health', xPlayer.getMeta('health') or GetEntityHealth(ped))
```

Comment:
Here it should be handled to get from xPlayer if the player is dead and not look at the native because they can use disconnect to resurrect.